tests/integration/cmd_link_unlink_test.go

```diff
@@ -9,138 +13,201 @@ import (
-				helper.DeleteDir(backendContext)
-			})
-			It("should fail", func() {
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "--s2i", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				helper.CopyExample(filepath.Join("source", "python"), backendContext)
-				helper.Cmd("odo", "create", "--s2i", "python", "backend", "--context", backendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-				stdErr := helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "1234").ShouldFail().Err()
-				Expect(stdErr).To(ContainSubstring("Unable to properly link to component backend using port 1234"))
-			})
+	When("two components are deployed", func() {
+		var frontendContext, backendContext, frontendURL, frontendComp, backendComp string
+		var runner helper.CliRunner
+
+		// checkDescribe: checks that the linked component and related variables are present in the output of odo describe
+		var checkDescribe = func(contextDir string, compName string, pushed bool, bindAsFiles bool) {
+			stdOut := helper.Cmd("odo", "describe", "--context", contextDir).ShouldPass().Out()
+			Expect(stdOut).To(ContainSubstring("Linked Services"))
+			Expect(stdOut).To(ContainSubstring(compName))
+			if pushed {
+				if bindAsFiles {
+					Expect(stdOut).To(ContainSubstring("/bindings"))
+				} else {
+					Expect(stdOut).To(ContainSubstring("SERVICE_BACKEND_IP"))
+					Expect(stdOut).To(ContainSubstring("SERVICE_BACKEND_PORT"))
+				}
+			}
+		}
+
+		// createAndPush: creates component, a URL for it and deploys it
+		var createAndPush = func(compType string, compName string, contextDir string) {
+			helper.CopyExample(filepath.Join("source", compType), contextDir)
+			helper.Cmd("odo", "create", compType, compName, "--context", contextDir, "--project", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "url", "create", "--port", "8080", "--context", contextDir).ShouldPass()
+			helper.Cmd("odo", "push", "--context", contextDir).ShouldPass()
+		}
+
+		JustBeforeEach(func() {
+			runner = helper.GetCliRunner()
+
+			frontendComp = fmt.Sprintf("frontend-%v", helper.RandString(3))
+			frontendContext = helper.CreateNewContext()
+			createAndPush("nodejs", frontendComp, frontendContext)
+			frontendURL = helper.DetermineRouteURL(frontendContext)
+
+			backendComp = fmt.Sprintf("backend-%v", helper.RandString(3))
+			backendContext = helper.CreateNewContext()
+			createAndPush("python", backendComp, backendContext)
+		})
+
+		JustAfterEach(func() {
+			helper.DeleteDir(frontendContext)
+			helper.DeleteDir(backendContext)
 		})
-	*/
 
-	/*
-		Context("When handling link/unlink between components", func() {
+		When("a link is created between the two components", func() {
 			JustBeforeEach(func() {
```

Comment:
https://github.com/openshift/odo/pull/4891#discussion_r685108407 Just to ensure that we are not abusing `JustBeforeEach` as per Ginkgo [Doc](https://onsi.github.io/ginkgo/) . So if we have options where we can avoid using `JustBeforeEach`, we should prefer to do that, no?

Reply:
Yes, that makes sense. Thanks for pointing it out! :)

Reply:
I've replaced all the JBE/JAE with BE/AE because it makes more sense in this context.